python/py-allennlp/Portfile

```diff
@@ -0,0 +1,77 @@
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           python 1.0
+
+github.setup        allenai allennlp 2.4.0 v
+revision            0
+name                py-${github.project}
+categories-append   textproc
+
+license             Apache-2
+maintainers         nomaintainer
+platforms           darwin
+
+description         An open-source NLP research library, built on PyTorch.
+
+long_description    An Apache 2.0 NLP research library, built on \
+                    PyTorch, for developing state-of-the-art deep \
+                    learning models on a wide variety of linguistic \
+                    tasks.
+
+homepage            https://www.allennlp.org/
+
+checksums           rmd160  acd8338381723cdddcb52136a5e70713529ea230 \
+                    sha256  3fbd02e6261d04fe2221e44c48cc936dbcefc11704c37b746ef7b96b705c9caa \
+                    size    2714266
+
+python.versions     38 39
+
+if {${name} ne ${subport}} {
+    depends_build-append \
+                    port:py${python.version}-setuptools
+
+    depends_run-append \
+                    port:py${python.version}-boto3 \
+                    port:py${python.version}-filelock \
+                    port:py${python.version}-h5py \
+                    port:py${python.version}-jsonnet \
+                    port:py${python.version}-jsonpickle \
+                    port:py${python.version}-lmdb \
+                    port:py${python.version}-more-itertools \
+                    port:py${python.version}-nltk \
+                    port:py${python.version}-numpy \
+                    port:py${python.version}-overrides \
+                    port:py${python.version}-pytest \
+                    port:py${python.version}-pytorch \
+                    port:py${python.version}-requests \
+                    port:py${python.version}-scikit-learn \
+                    port:py${python.version}-scipy \
+                    port:py${python.version}-sentencepiece \
+                    port:py${python.version}-spacy \
+                    port:py${python.version}-tensorboardX \
+                    port:py${python.version}-torchvision \
+                    port:py${python.version}-tqdm \
+                    port:py${python.version}-transformers
+
+    depends_test-append \
+                    port:py${python.version}-black \
+                    port:py${python.version}-codecov \
+                    port:py${python.version}-coverage \
+                    port:py${python.version}-flake8 \
+                    port:py${python.version}-flaky \
+                    port:py${python.version}-matplotlib \
+                    port:py${python.version}-mypy \
+                    port:py${python.version}-pytest \
+                    port:py${python.version}-pytest-benchmark \
+                    port:py${python.version}-pytest-cov \
+                    port:py${python.version}-responses
+
+    test.run        yes
+    test.env        PYTHONPATH=${worksrcpath}/build/lib
```

Comment:
sorry, one more question... The Portfile doesn't set `supported_archs noarch`, so supposedly some code gets compiled here - correct? If so, shouldn't you instead set:
```
    pre-test {
        test.env    PYTHONPATH=[glob -nocomplain ${worksrcpath}/build/lib*]
    }
``` 

did you confirm that the test-suite runs by running the tests without having the port already installed?

Reply:
Yes, I did. The port install some `cpython` stuff, but I don't know if the `supported_archs` should be set.

Reply:
@reneeotten I added `supported_archs noarch`. I believe that this is ready to merge.